stubs/redis/redis/cluster.pyi

```diff
@@ -1,75 +1,79 @@
-from _typeshed import Incomplete
+from _typeshed import Incomplete, Self
 from collections.abc import Callable
 from threading import Lock
+from types import TracebackType
 from typing import Any, ClassVar, Generic
 
-from redis.client import PubSub
-from redis.commands import RedisClusterCommands
+from redis.client import PubSub, Redis
+from redis.commands import CommandsParser, RedisClusterCommands
 from redis.commands.core import _StrType
-from redis.connection import BaseParser, Encoder
+from redis.connection import BaseParser, Connection, Encoder
 from redis.exceptions import RedisError
 
-def get_node_name(host, port): ...
-def get_connection(redis_node, *args, **options): ...
+def get_node_name(host: str, port: int) -> str: ...
+def get_connection(redis_node: Redis[Any], *args, **options) -> Connection: ...
 def parse_scan_result(command, res, **options): ...
 def parse_pubsub_numsub(command, res, **options): ...
 def parse_cluster_slots(resp, **options): ...
 
 PRIMARY: str
 REPLICA: str
 SLOT_ID: str
-REDIS_ALLOWED_KEYS: Any
-KWARGS_DISABLED_KEYS: Any
-READ_COMMANDS: Any
+REDIS_ALLOWED_KEYS: tuple[str, ...]
+KWARGS_DISABLED_KEYS: tuple[str, ...]
 
 def cleanup_kwargs(**kwargs): ...
 
 # It uses `DefaultParser` in real life, but it is a dynamic base class.
 class ClusterParser(BaseParser): ...
 
-class RedisCluster(RedisClusterCommands[_StrType], Generic[_StrType]):
+class AbstractRedisCluster:
     RedisClusterRequestTTL: ClassVar[int]
     PRIMARIES: ClassVar[str]
     REPLICAS: ClassVar[str]
     ALL_NODES: ClassVar[str]
     RANDOM: ClassVar[str]
     DEFAULT_NODE: ClassVar[str]
     NODE_FLAGS: ClassVar[set[str]]
-    COMMAND_FLAGS: ClassVar[Any]
+    COMMAND_FLAGS: ClassVar[dict[str, str]]
     CLUSTER_COMMANDS_RESPONSE_CALLBACKS: ClassVar[dict[str, Any]]
-    RESULT_CALLBACKS: ClassVar[Any]
+    RESULT_CALLBACKS: ClassVar[dict[str, Any]]
     ERRORS_ALLOW_RETRY: ClassVar[tuple[type[RedisError], ...]]
-    user_on_connect_func: Any
-    encoder: Any
-    cluster_error_retry_attempts: Any
-    command_flags: Any
-    node_flags: Any
-    read_from_replicas: Any
+
+class RedisCluster(AbstractRedisCluster, RedisClusterCommands[_StrType], Generic[_StrType]):
+    user_on_connect_func: Callable[[Connection], object]
+    encoder: Encoder
+    cluster_error_retry_attempts: int
+    command_flags: dict[str, str]
+    node_flags: set[str]
+    read_from_replicas: bool
     reinitialize_counter: int
     reinitialize_steps: int
-    nodes_manager: Any
-    cluster_response_callbacks: Any
-    result_callbacks: Any
-    commands_parser: Any
-    def __init__(
+    nodes_manager: NodesManager
+    cluster_response_callbacks: dict[str, Any]
+    result_callbacks: dict[str, Any]
```

Comment:
```suggestion
    result_callbacks: CaseInsensitiveDict[str, Callable[[Incomplete, Incomplete], Incomplete]]
```

Reply:
Why has this one been marked "resolved" but not been applied? :)

Reply:
Now it is! 👍

Reply:
Thanks!